src/systemctl/systemctl.c

```diff
@@ -7264,7 +7268,58 @@ static int create_edit_temp_file(const char *new_path, const char *original_path
         if (r < 0)
                 return log_error_errno(r, "Failed to create directories for \"%s\": %m", new_path);
 
-        r = copy_file(original_path, t, 0, 0644, 0, 0, COPY_REFLINK);
+        if (original_path) {
+                r = copy_file(original_path, t, 0, 0644, 0, 0, COPY_REFLINK);
+        } else {
```

Comment:
according to our usually coding style, please no {} around single-line if blocks